### ISSUE TITLE
[🐛 Bug]: browser in a custom location is not found in 4.11.1 (Javascript)

### ISSUE THREAD

**Issue by fippo** [SeleniumHQ/selenium#12604]
### What happened?

We started seeing CI failures in the webrtc samples repository.
This boiled down to a dependency version bump of selenium-webdriver (JS) from 4.10.0 to 4.11.0 (or .1)

https://github.com/webrtc/samples/issues/1614
has the full history including several CI runs. I've pinned the version to 4.10.0 for the time being which resolved the immediate issue.

### How can we reproduce the issue?
I've created a minimal reproduction:
https://github.com/fippo/selenium-not-found
It will require linux to reproduce since that is the only platform supported by travis-multirunner (which is used to install a custom Chrome)

To reproduce,

```shell
* run npm install (which will currently install 4.11.1 and fail; change to 4.10.0 as required)
* set BROWSER=chrome and BVER=unstable
* run node_mode/travis-multirunner/setup.sh
* optionally verify this makes Chrome accessible at browsers/bin/chrome-unstable
* run node webdriver.js
```
Using 4.10.0 this works as expected, shows chrome://version (which should be M118 currently) and closes.
Starting with 4.11.0 this fails to find the browser in the custom location that is set via `setChromeBinaryPath`

Prior to 4.11.1 it did not show this line:
```
Selenium Manager binary found at 
    /home/runner/work/samples/samples/node_modules/selenium-webdriver/bin/linux/selenium-manager
```
It looks like 4.11.1 started using selenium-manager.

### Relevant log output

```shell
WebDriverError: unknown error: no chrome binary at "browsers/bin/chrome-unstable"
```

### Operating System

Ubuntu

### Selenium version

Javascript 4.11.1

### What are the browser(s) and version(s) where you see this issue?

Chrome

### What are the browser driver(s) and version(s) where you see this issue?

automatically installed

### Are you using Selenium Grid?

no

**Comment by diemol** [comment]
Selenium Manager is now used to check the path and see if ChromeDriver is available. The issue comes when Selenium Manager tries to check the browser version to find the right ChromeDriver (I think). The browser binary path seems to be a relative path, is it possible to give an absolute path?

Please run again with debug logging and share the logs https://www.selenium.dev/documentation/webdriver/troubleshooting/logging/

Also, something else to try is to remove:
```
      .setChromeService(
          new chrome.ServiceBuilder().addArguments('--disable-build-check')
```

**Comment by fippo** [comment]
Code change are here: https://github.com/fippo/selenium-not-found/commit/862ed856e25652ed29a057937dd8ee71951117f4

No change with eithe removing the disable-build-check nor setting an absolute path (or manually resolving the symlinked path)

With logging enabled this is the output:
```
LOGGER Logger {
  name_: 'webdriver',
  level_: Level { name_: 'ALL', value_: 0 },
  parent_: Logger {
    name_: '',
    level_: Level { name_: 'OFF', value_: Infinity },
    parent_: null,
    handlers_: null
  },
  handlers_: null
}
Selenium Manager binary found at /home/username/selenium-repro/node_modules/selenium-webdriver/bin/linux/selenium-manager
Driver path: /home/username/.cache/selenium/chromedriver/linux64/118.0.5966.0/chromedriver
Browser path: "/home/username/selenium-repro/node_modules/travis-multirunner/browsers/chrome/unstable/chrome"
[2023-08-24T12:40:26Z] [FINEST] [webdriver.http] Translating command: getStatus
[2023-08-24T12:40:26Z] [FINEST] [webdriver.http] Building HTTP request: {"method":"GET","path":"/status"}
[2023-08-24T12:40:26Z] [FINER] [webdriver.http.Executor] >>> GET /status
[2023-08-24T12:40:26Z] [FINEST] [webdriver.http] Translating command: newSession
[2023-08-24T12:40:26Z] [FINEST] [webdriver.http] Building HTTP request: {"method":"POST","path":"/session"}
[2023-08-24T12:40:26Z] [FINER] [webdriver.http.Executor] >>> POST /session
[2023-08-24T12:40:26Z] [FINER] [webdriver.http.Executor] >>>
GET /status HTTP/1.1
accept: application/json; charset=utf-8

{}
<<<
HTTP/1.1 200
content-length: 246
content-type: application/json; charset=utf-8
cache-control: no-cache
connection: close

{"value":{"build":{"version":"118.0.5966.0 (a8e03c20d55aff8935c0460d210be0f20720f8e9-refs/branch-heads/5966@{#1})"},"message":"ChromeDriver ready for new sessions.","os":{"arch":"x86_64","name":"Linux","version":"6.2.0-26-generic"},"ready":true}}
[2023-08-24T12:40:26Z] [FINER] [webdriver.http.Executor] >>>
POST /session HTTP/1.1
accept: application/json; charset=utf-8

{"capabilities":{"firstMatch":[{}],"alwaysMatch":{"browserName":"chrome","goog:chromeOptions":{"binary":"\"/home/username/selenium-repro/node_modules/travis-multirunner/browsers/chrome/unstable/chrome\""}}}}
<<<
HTTP/1.1 500
content-length: 797
content-type: application/json; charset=utf-8
cache-control: no-cache

{"value":{"error":"unknown error","message":"unknown error: no chrome binary at \"/home/username/selenium-repro/node_modules/travis-multirunner/browsers/chrome/unstable/chrome\"","stacktrace":"#0 0x5623c8c6abd3 \u003Cunknown>\n#1 0x5623c893ff57 \u003Cunknown>\n#2 0x5623c897042f \u003Cunknown>\n#3 0x5623c896f6aa \u003Cunknown>\n#4 0x5623c89b1e86 \u003Cunknown>\n#5 0x5623c89a8ad3 \u003Cunknown>\n#6 0x5623c897b71b \u003Cunknown>\n#7 0x5623c897c50e \u003Cunknown>\n#8 0x5623c8c30e48 \u003Cunknown>\n#9 0x5623c8c34be0 \u003Cunknown>\n#10 0x5623c8c3f05c \u003Cunknown>\n#11 0x5623c8c357f8 \u003Cunknown>\n#12 0x5623c8c02c5f \u003Cunknown>\n#13 0x5623c8c59258 \u003Cunknown>\n#14 0x5623c8c59424 \u003Cunknown>\n#15 0x5623c8c69d63 \u003Cunknown>\n#16 0x7facdf894b43 \u003Cunknown>\n"}}
/home/username/selenium-repro/node_modules/selenium-webdriver/lib/error.js:524
    let err = new ctor(data.message)
              ^

WebDriverError: unknown error: no chrome binary at "/home/username/selenium-repro/node_modules/travis-multirunner/browsers/chrome/unstable/chrome"
    at Object.throwDecodedError (/home/username/selenium-repro/node_modules/selenium-webdriver/lib/error.js:524:15)
    at parseHttpResponse (/home/username/selenium-repro/node_modules/selenium-webdriver/lib/http.js:601:13)
    at Executor.execute (/home/username/selenium-repro/node_modules/selenium-webdriver/lib/http.js:529:28)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  remoteStacktrace: '#0 0x5623c8c6abd3 <unknown>\n' +
    '#1 0x5623c893ff57 <unknown>\n' +
    '#2 0x5623c897042f <unknown>\n' +
    '#3 0x5623c896f6aa <unknown>\n' +
    '#4 0x5623c89b1e86 <unknown>\n' +
    '#5 0x5623c89a8ad3 <unknown>\n' +
    '#6 0x5623c897b71b <unknown>\n' +
    '#7 0x5623c897c50e <unknown>\n' +
    '#8 0x5623c8c30e48 <unknown>\n' +
    '#9 0x5623c8c34be0 <unknown>\n' +
    '#10 0x5623c8c3f05c <unknown>\n' +
    '#11 0x5623c8c357f8 <unknown>\n' +
    '#12 0x5623c8c02c5f <unknown>\n' +
    '#13 0x5623c8c59258 <unknown>\n' +
    '#14 0x5623c8c59424 <unknown>\n' +
    '#15 0x5623c8c69d63 <unknown>\n' +
    '#16 0x7facdf894b43 <unknown>\n'
}
```
This also reproduces when just setting the Chrome binary path to the system-wide `/usr/bin/chrome`, i.e. without travis-multirunner.

Does selenium-manager chroot itself?

**Comment by diemol** [comment]
This is strange because the message comes from ChromeDriver, saying that a binary was not found. 

What ChromeDriver version are you using when Selenium 4.10 is used?

**Comment by fippo** [comment]
Reproduced with both 98 (terribly old) and 116.

But I found a new hint by logging `args` here
https://github.com/SeleniumHQ/selenium/blob/trunk/javascript/node/selenium-webdriver/common/seleniumManager.js#L83
then removing the quotes fixed it and and this turns out to be fixed by https://github.com/SeleniumHQ/selenium/commit/ec2f0fb156259c0cb18f09e5031506f245b5fa37 which fixes #12479 already :tada:

Thanks for the help! I'll wait for the next release
